api/apiroutes.go

```diff
@@ -743,6 +744,24 @@ func (c *appContext) getDecodedTx(w http.ResponseWriter, r *http.Request) {
 	writeJSON(w, tx, m.GetIndentCtx(r))
 }
 
+func (c *appContext) getTxSwapInfo(w http.ResponseWriter, r *http.Request) {
+	txid, err := m.GetTxIDCtx(r)
+	if err != nil {
+		http.Error(w, http.StatusText(422), 422)
+		return
+	}
+
+	tx, err := c.DataSource.GetAtomicSwapInfo(txid.String())
```

Comment:
`appContext` has it's own node client, so it could call the hypothetical rpcutils function instead.

Reply:
Oh, well this one is still DataSource I guess since it needs SpendingInfo, or it could follow that logic her.e